rsciio/trivista/_api.py

```diff
@@ -0,0 +1,745 @@
+        )
+        infoSerialized = _etree_to_dict(et_infoSerialized, only_top_lvl=False)
+        infoSerialized_processed = _process_info_serialized(
+            infoSerialized["Info"]["Groups"]["Group"]
+        )
+        self.original_metadata["Document"]["InfoSerialized"] = infoSerialized_processed
+
+        ## Hardware section
+        metadata_head = et_root[0]
+        metadata_hardware = _etree_to_dict(metadata_head, only_top_lvl=False)
+        if not filter_original_metadata:
+            self.unfiltered_original_metadata = deepcopy(self.original_metadata)
+            self.unfiltered_original_metadata.update(deepcopy(metadata_hardware))
+
+        ## filter LightSources section (Laser) via wavelength
+        for laser in metadata_hardware["Hardware"]["LightSources"]["LightSource"]:
+            try:
+                calibration_wl = float(
+                    infoSerialized_processed["Calibration"]["Laser_Wavelength"]
+                )
+                laser_wl = float(laser["Wavelengths"]["Value_0"])
+            except KeyError:
+                pass
+            else:
+                if np.isclose(calibration_wl, laser_wl) and not np.isclose(laser_wl, 0):
+                    metadata_hardware["Hardware"]["LightSources"]["LightSource"] = laser
+
+        ## filter Detector section via "name"
+        for detector in metadata_hardware["Hardware"]["Detectors"]["Detector"]:
+            if detector["Name"] == infoSerialized_processed["Detector"]["Name"]:
+                metadata_hardware["Hardware"]["Detectors"]["Detector"] = detector
+
+        ## filter microscope section (objective) via isEnabled tag
+        for microscope in metadata_hardware["Hardware"]["Microscopes"]["Microscope"]:
+            for objective in microscope["Objectives"]["Objective"]:
+                if objective["IsEnabled"] == "True":
+                    metadata_hardware["Hardware"]["Microscopes"][
+                        "Microscope"
+                    ] = microscope
+                    metadata_hardware["Hardware"]["Microscopes"]["Microscope"][
+                        "Objectives"
+                    ]["Objective"] = objective
+
+        ## get serialnumbers for all used spectrometers
+        ## contrary to the other parts, multiple spectrometers can be used
+        spectrometer_serial_numbers = []
+        spectrometer_serialized_list = []
+        for key, val in infoSerialized_processed["Spectrometers"].items():
+            spectrometer_serial_numbers.append(val["Serialnumber"])
+            spectrometer_serialized_list.append(key)
+
+        ## filter spectrometers via serialnumber
+        ## result for one spectrometer:
+        ## Spectrometers
+        ##     - Spectrometer
+        ##         - ...
+        ## result for 2 spectrometers:
+        ## Spectrometers
+        ##     - Spectrometer1
+        ##         - ...
+        ##     - Spectrometer2
+        ##         - ...
```

Comment:
I don't see the point in having a different naming convention for 1 or multiple spectrometer. It creates confusion compared to having `Spectrometer.Spectrometer1` for only 1 spectro in my opinion (and also starting numbering from 1 and not 0 is a bit odd but after all why not).

Reply:
When only 1 Spectrometer is used, I wanted that the format of ``metadata`` is consistent with LumiSpy (Acquisition_instrument/Spectrometer/...). Using the same format for ``original_metadata`` makes the mapping more convenient.

Reply:
Fair enough.